### PR TITLE
googlemanagedprometheus exporter initial commit

### PR DESCRIPTION
**Pull request by dashpole**
**Description:**

Part of https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/10543.

Add structure for the new GMP exporter.

This also fixes a bug in the module checker.

**Documentation:**

New Readme

### CODE REVIEW

File: .github/workflows/scripts/check-collector-module-version.sh [hunk 1]
```diff
@@ -49,8 +49,10 @@ check_collector_versions_correct() {
          version=$(echo "$mod_line" | cut -d" " -f2)
 
          # To account for a module on its own 'require' line,
-         # the version field is shifted right by 1
-         if [ "$version" == "$collector_module" ]; then
+         # the version field is shifted right by 1. Match
+         # with or without a trailing space at the end to account
+         # for the space at the end of some collector modules
+         if [ "$version" == "$collector_module" ] || [ "$version " == "$collector_module" ]; then
```
Comment (mx-psi) [inline .github/workflows/scripts/check-collector-module-version.sh:55]: Sounds like this should go on a different PR? Also cc @hickeyma

File: exporter/googlemanagedprometheusexporter/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# Google Managed Service for Prometheus Exporter
+
+| Status                   |                       |
+| ------------------------ |-----------------------|
+| Stability                | [alpha]               |
```
Comment (mx-psi) [inline exporter/googlemanagedprometheusexporter/README.md:5]: The link is missing. Also, should this be 'in development' instead?

File: .github/workflows/scripts/check-collector-module-version.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (hickeyma) [reply .github/workflows/scripts/check-collector-module-version.sh:55]: Nice catch, thanks for finding this @dashpole

Comment (hickeyma) [reply .github/workflows/scripts/check-collector-module-version.sh:55]: I agree with @mx-psi. It's probably best put this in a separate PR for better trace afterwards of why the change. Do you want to push the PR @dashpole or if not, I can do it?

Comment (hickeyma) [reply .github/workflows/scripts/check-collector-module-version.sh:55]: Ok, I'll push fix shortly,

Comment (hickeyma) [reply .github/workflows/scripts/check-collector-module-version.sh:55]: Pushed #10885 for the fix.

[the remark this answers is not in the document]
Comment (dashpole) [reply .github/workflows/scripts/check-collector-module-version.sh]: Thanks :).  I removed this fix from this PR.

File: exporter/googlemanagedprometheusexporter/README.md (hunk 2 above)
Comment (dashpole) [reply exporter/googlemanagedprometheusexporter/README.md]: changed to in development

Comment (dashpole) [reply exporter/googlemanagedprometheusexporter/README.md]: added link as well

**Review by mx-psi — APPROVED** [review]

**Comment by dashpole** [comment]
The `check-collector-module-version` check will fail until this is rebased on https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/10885.
